src/clients/ntp-client_proposal.rb

```diff
@@ -89,56 +89,56 @@ def main
     end
 
     def ui_help_text
-      # help text
-      tmp = _(
-        "<p>Press <b>Synchronize Now</b>, to get your system time set correctly " \
-        "using the selected NTP server. If you want to make use of NTP permanently, " \
-        "enable the <b>Save NTP Configuration</b> option</p>"
-      )
-
-      tmp = Ops.add(
-        tmp,
-        _(
-          "<p>Enabling <b>Run NTP as daemon</b> option, the NTP service will be " \
-          "started as daemon. Otherwise the system time will be synchronized periodically. " \
-          "The default interval is 15 min. You can change it after installation " \
-          "with the <b>yast2 ntp-client module</b>.</p>"
+      if Stage.initial
+        # help text
+        tmp = _(
+          "<p>Press <b>Synchronize Now</b>, to get your system time set correctly " \
+          "using the selected NTP server. If you want to make use of NTP permanently, " \
+          "enable the <b>Save NTP Configuration</b> option</p>"
         )
-      )
 
-      # help text, cont.
-      if !Stage.initial
         tmp = Ops.add(
           tmp,
           _(
-            "<p>Using the <b>Configure</b> button, open the advanced NTP configuration.</p>"
+            "<p>Enabling <b>Run NTP as daemon</b> option, the NTP service will be " \
+            "started as daemon. Otherwise the system time will be synchronized periodically. " \
+            "The default interval is 15 min. You can change it after installation " \
+            "with the <b>yast2 ntp-client module</b>.</p>"
+          )
+        )
+        # help text, cont.
+        tmp = Ops.add(
+          tmp,
+          _(
+            "<p>Synchronization with the NTP server can be done only when " \
+            "the network is configured.</p>"
           )
         )
+      else
+        # help text
+        tmp = _(
+          "<p>Using the <b>Configure</b> button, open the advanced NTP configuration.</p>"
+        )
       end
 
-      # help text, cont.
-      tmp = Ops.add(
-        tmp,
-        _(
-          "<p>Synchronization with the NTP server can be done only when " \
-          "the network is configured.</p>"
-        )
-      )
       tmp
     end
 
     def ui_enable_disable_widgets(enabled)
       UI.ChangeWidget(Id(:ntp_address), :Enabled, enabled) if select_ntp_server
-      UI.ChangeWidget(Id(:run_service), :Enabled, enabled)
-      # FIXME: With chronyd, we cannot synchronize if the service is already
-      # running, we could force a makestep in this case, but then the button
-      # should be reworded and maybe the user should confirm it (bsc#1087048)
-      if !NetworkService.isNetworkRunning || Service.Active(NtpClient.service_name)
-        UI.ChangeWidget(Id(:ntp_now), :Enabled, false)
-      else
-        UI.ChangeWidget(Id(:ntp_now), :Enabled, enabled)
+      if Stage.initial
```

Comment:
BTW when seeing all this conditions I start thinking that it would be nicer to have own class for runtime and installation. I know it is not possible for maintenance branch, but in general it can make code much cleaner as now runtime and installation differs a lot.

Reply:
Yes, we have already opened a new card for refactoring the complete code.